### DISCUSSION TITLE
Honouring RI (referential integrity) in generated data

### DISCUSSION THREAD

**Discussion by jsmith085** [getsynth/synth#321] [Q&A] [answered]
Hello Team Synth!

Your project looks very interesting.......I'm wondering if Synth can help generate data such that it honours the referential integrity relationships between entities?  For example, customers can have 0 or many orders and each order has 1 or many order_items.  Can these types of relationship be defined and data generated....ideally based on some sort of discovery over the target schema objects (table, primary and foreign keys and indexes defined)?

Cheers and thanks advance 

Jason Smith

**Comment by christos-h** [comment, marked as answer]
Hey @jsmith085 - `synth import` will automatically infer foreign keys. Constraints such as 'at least one' or 'zero or more' need to be defined explicitly with relative collection sizes (see -> https://www.getsynth.com/docs/getting_started/schema).

FK constraints can also be specified manually by using the `same_as` generator, see -> https://www.getsynth.com/docs/content/same-as

Hope this helps!

**Reply by dan-schofer** [reply]
@christos-h is it possible to configure a foreign key relationship between synth schemas in different namespaces? (tables in different schemas in my postgres instance)
